packages/eslint-plugin/src/rules/array-type.ts

```diff
@@ -99,12 +103,20 @@ export default util.createRule<Options, MessageIds>({
     messages: {
       errorStringGeneric:
         "Array type using '{{type}}[]' is forbidden. Use 'Array<{{type}}>' instead.",
-      errorStringGenericSimple:
-        "Array type using '{{type}}[]' is forbidden for non-simple types. Use 'Array<{{type}}>' instead.",
+      errorStringReadonlyGeneric:
+        "Array type using 'readonly {{type}}[]' is forbidden. Use 'ReadonlyArray<{{type}}>' instead.",
       errorStringArray:
         "Array type using 'Array<{{type}}>' is forbidden. Use '{{type}}[]' instead.",
+      errorStringReadonlyArray:
+        "Array type using 'ReadonlyArray<{{type}}>' is forbidden. Use 'readonly {{type}}[]' instead.",
       errorStringArraySimple:
         "Array type using 'Array<{{type}}>' is forbidden for simple types. Use '{{type}}[]' instead.",
+      errorStringReadonlyArraySimple:
+        "Array type using 'ReadonlyArray<{{type}}>' is forbidden for simple types. Use 'readonly {{type}}[]' instead.",
+      errorStringGenericSimple:
+        "Array type using '{{type}}[]' is forbidden for non-simple types. Use 'Array<{{type}}>' instead.",
+      errorStringReadonlyGenericSimple:
+        "Array type using 'readonly {{type}}[]' is forbidden for non-simple types. Use 'ReadonlyArray<{{type}}>' instead.",
```

Comment:
This is a lot of message IDs but the new difference is writing Array vs `ReadonlyArray`. How about using something like `{{className}}` instead of several new message IDs?

```ts
"Array type using 'readonly {{type}}[]' is forbidden for non-simple types. Use '{{className}}<{{type}}>' instead."
```

Reply:
Fixed 4c7aaf9d7b8cbf336822c263f52d5e3f4909ed83